### DISCUSSION TITLE
USB or another wired method

### DISCUSSION THREAD

**Discussion by kynerav** [tbnobody/OpenDTU#1739] [Q&A]
Hi
Is there a way to get the API but via USB or I2C or something else hardwired?

I use Wemos D1 mini + modbus smart meters for zero export and to control the dimmer to increase water heater power depending on the "free" energy. It works perfect but wifi router link is most vulnerable and unstable in this case. Especially after power off.
It would be good to use http for monitoring but to make the limiter work hard.

Any other way to get/set data?

**Comment by MetaChuh** [comment]
as far as i know, commands and queries are only via wifi or connected ethernet stack.
no serial, usb to serial, or web serial commands exist.
at least that i know of.
